single_kernel_mongo/managers/ldap.py

```diff
@@ -113,9 +128,7 @@ def on_certificate_available(self, certificate: str, ca: str, chain: list[str]):
         full_chain = "\n".join(chain)
         self.dependent.save_ca_cert_to_trust_store(TrustStoreFiles.LDAP, full_chain)
 
-        if self.state.ldap.is_ready():
-            logger.info("Restarting mongodb server for LDAP integration")
-            self.dependent.restart_charm_services()
+        self.restart_when_ready()
```

Comment:
Are there any dangers of not deferring when not ready?

Reply:
We specifically don't want to defer here:
1) we are waiting on two relations
2) restart when ready first gets the component status:
 * if there is no status, there is no integration so we don't restart
 * if there is an active status, then everything is integrated and we have checked that the connection works
 * Else, there is something blocking and we go in blocked state because this will not auto resolve.